pybaseball/fangraphs_projections.py

```diff
@@ -0,0 +1,112 @@
+import json
+
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
+from pybaseball.teamid_lookup import fg_team_id_dict
+
+# pylint: disable=line-too-long
+_URL = "https://www.fangraphs.com/projections?pos={pos}&stats={stats}&type={proj_source}&team={team}&lg={lg}"
+
+
+def fg_projections(proj_source: str = "zips", position: str = "batters", league: str = "mlb", team: str = "") -> pd.DataFrame:
+    """Retrieves player projection statistics from Fangraphs. View markdown documentation for complete list of valid arguments.
+
+    Args:
+        proj_source (str, optional): Projection source, including pre-season, rest of season, updated in-season, 3 year, and on pace leader options. Defaults to "zips".
+        position (str, optional): Batters/Pitchers, field position, or pitcher type. Defaults to "batters".
+        league (str, optional): mlb, al, or nl. Defaults to "mlb".
+        team (str, optional): filter to a specific team by team abbreviation (i.e. for the Philadelphia Phillies, use `PHI`). Defaults to empty (all teams)
+
+    Returns:
+        pd.DataFrame: Projections data set given applied filters, columns vary for batters versus pitcher. Returns empty dataframe when no data found on Fangraphs
+    """
+    args = _FgProjectionArgument(position, proj_source, league, team)
+    source = requests.get(args.url, timeout=None)
+
+    soup = BeautifulSoup(source.content, 'html.parser')
+    raw = soup.find('script', id='__NEXT_DATA__', type='application/json').string
+    data = json.loads(raw)['props']['pageProps']['dehydratedState']['queries'][0]['state']['data']
+    result = pd.read_json(json.dumps(data))
```

Comment:
Turns out the entire projection payload is present in the page source even if the default view only displays the first 30 rows

Reply:
I think you're doing too much work here there's an api: 
pitch_df = pd.DataFrame(json.loads(requests.get('[https://www.fangraphs.com/api/projections?stats=pit&type=steamer').content]

Reply:
Per @blacktj we may want to utilize non-public but otherwise accessible API to capture this data directly: 
```
json.loads(requests.get('https://www.fangraphs.com/api/projections?stats=pit&type=steamer').content
```

Reply:
> 

You just saved me $10/month and made my life a ton easier.